GliaWidgets/Lib/Command/Command.swift

```diff
@@ -0,0 +1,67 @@
+/// Developer-friendly wrapper around a closure.
+/// It makes debugging easier by providing callee information.
+/// Note: since closure does not conform to `Equatable`, it will
+/// also be ignored in `Equatable` and `Hashable` implementations
+/// of `Command`.
+///
+/// Example of declaration and execution:
+/// ```
+/// let validateEmail = Command<String> { email in /* email validation goes here */ }
+/// validateEmail("email@example.test")
+/// ```
+struct Command<T>: Hashable {
+    let tag: String
+    let file: String
+    let function: String
+    let line: UInt
+    let closure: (T) -> Void
+
+    init(
+        tag: String = "",
+        file: StaticString = #file,
+        function: StaticString = #function,
+        line: UInt = #line,
+        closure: @escaping (T) -> Void
+    ) {
+        self.tag = tag
+        self.file = "\(file)"
+        self.function = "\(function)"
+        self.line = line
+        self.closure = closure
+    }
+
+    func execute(with value: T) {
+        closure(value)
+    }
+
+    func callAsFunction(_ value: T) {
```

Comment:
I don't actually understand what is the difference between this method and `execute(with value: T)`. Could you please clarify?

Reply:
Sure: it allows code like this:
```
let validateEmail = Command<String> { email in /* email validation goes here */ }
validateEmail("email@example.test")
```
It makes usage more ergonomic by omitting the call to `execute`. Without it the code would be like this:
```
validateEmail.execute(with: "email@example.test")
```

Reply:
Ohh, I see. I totally forgot this `callAsFunction` feature)